oars_gb_pkg/helpers/path_planning/short_course_planner.py

```diff
@@ -0,0 +1,145 @@
+from turtle import *
+from time import sleep
+
+class ShortCoursePlanner:
+
+    def polar_effeciency(self, windangle, alpha):
+        '''
+        Calculates the speed for a certain angle
+        :param windangle: the wind direction
+        :param alpha: angle being tested (relative to the wind angle)
+        :return: the vector components for the velocity for the angle
+        '''
+        # Convert angle to radians
+        angle = (windangle + alpha) * 180
+        alpha *= np.pi / 180
+
+        # Calculate magnitude using the polar effeciency function
+        mag = (1 - np.cos(alpha)) * (1 + 0.3 * np.cos(alpha)) / (1 - 0.5 * np.cos(alpha))
+
+        # Return vector components
+        return [mag * np.cos(angle), mag * np.sin(angle)]
+
+    # finds the maximum velocity and angle that can be achieved for each value of alpha for the right side and the left side
+    def optimum(self, windangle, path):
+        '''
+            :param wind: angle representing the wind
+            :param path: vector representing the path taken to reach from boat's position to the target position
+            :return:
+            A tuple containing the following information:
+            windangle_max_R - max boat heading on right side
+            windangle_max_L - max boat heading on left side
+            vt_max_R - max velocity on right side
+            vt_max_L - max velocity on left side
+            '''
+
+        # Initializing the right side and left side optimum wind angles
+        windangle_max_R = windangle
+        windangle_max_L = windangle
+
+        # Initializing the right side and left side optimum max velocities
+        vt_max_R = 0
+        vt_max_L = 0
+
+        # Initializes alpha, or the angle being tested (relative to the wind direction)
+        alpha = 0
+
+        # Checks all alphas between 0 and 180 and calculates the theoretical speeds
+        # to find an optimal heading on the right hand side
+        for alpha in range(180):
+
+            # Uses the polar effeciency function to calculcate the hypothetical velocity of the boat at that angle
+            vb_hyp = self.polar_effeciency(windangle, alpha)
+
+            # Projecting the max speed in the direction of the path
+            vt_test = np.dot(vb_hyp,path)
+
+            # Check if this velocity is the new max
+            if(vt_test>vt_max_R):
+                vt_max_R = vt_test
+                windangle_max_R = (windangle + alpha) % 360
+
+        # Analogous to the right side, but for the left side
+        for alpha in range (-180,0):
+            vb_hyp = self.polar_effeciency(windangle, alpha)
+            vt_test = np.dot(vb_hyp, path)
+            if (vt_test>vt_max_L):
+                vt_max_L = vt_test
+                windangle_max_L = (windangle + alpha) % 360
+
+        return vt_max_R,vt_max_L,windangle_max_R,windangle_max_L
+
+
+
+    def get_new_dir(self, vt_max_R, vt_max_L, windangle_max_R, windangle_max_L, p_c, path, boat_heading):
```

Comment:
Is this determining the best direction? Perhaps `_get_best_dir` would be more descriptive. :+1: for including `get` in it already though.